Spigot-Server-Patches/0546-Configurable-dungeon-seed.patch

```diff
@@ -0,0 +1,78 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: William Blake Galbreath <Blake.Galbreath@GMail.com>
+Date: Wed, 15 Jul 2020 12:40:25 -0500
+Subject: [PATCH] Configurable dungeon seed
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index 741c06a27e664211a7023a6369f8e69cbb41a321..9b3835ef0f7e796f2a45f3a8afe112f89cb9ab30 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -435,4 +435,9 @@ public class PaperConfig {
+         allowPistonDuplication = getBoolean("settings.unsupported-settings.allow-piston-duplication", config.getBoolean("settings.unsupported-settings.allow-tnt-duplication", false));
+         set("settings.unsupported-settings.allow-tnt-duplication", null);
+     }
++
++    public static int dungeonSeed = -1;
++    private static void seedSettings() {
++        dungeonSeed = getInt("settings.seed.dungeon", dungeonSeed);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/WorldGenDecoratorDungeon.java b/src/main/java/net/minecraft/server/WorldGenDecoratorDungeon.java
+index 70d222996016e33bdf4d6d5dacf5a58eb3380d34..0e4ab749de697dbc712dfbf30b6540034d633c53 100644
+--- a/src/main/java/net/minecraft/server/WorldGenDecoratorDungeon.java
++++ b/src/main/java/net/minecraft/server/WorldGenDecoratorDungeon.java
+@@ -6,6 +6,7 @@ import java.util.stream.IntStream;
+ import java.util.stream.Stream;
+ 
+ public class WorldGenDecoratorDungeon extends WorldGenDecorator<WorldGenDecoratorDungeonConfiguration> {
++    private Random random; // Paper
+ 
+     public WorldGenDecoratorDungeon(Codec<WorldGenDecoratorDungeonConfiguration> codec) {
+         super(codec);
+@@ -14,10 +15,16 @@ public class WorldGenDecoratorDungeon extends WorldGenDecorator<WorldGenDecorato
+     public Stream<BlockPosition> a(GeneratorAccess generatoraccess, ChunkGenerator chunkgenerator, Random random, WorldGenDecoratorDungeonConfiguration worldgendecoratordungeonconfiguration, BlockPosition blockposition) {
+         int i = worldgendecoratordungeonconfiguration.b;
+ 
++        if (this.random == null) {
++            int seed = com.destroystokyo.paper.PaperConfig.dungeonSeed;
```

Comment:
This should be stored on world for the instance and then just create local as world version or existing local without any field additions as this instance is concurrently accessed and different worlds can have different seeds